### ISSUE TITLE
Default process group has not been initialized, please make sure to call init_process_group.

### ISSUE THREAD

**Issue by 13331549708** [NVlabs/FB-BEV#42]
When I use launch.json in Vscode to debug the code, I am training with a single card, and it will report an error: Default process group has not been initialized, please make sure to call init_process_group.
Can anyone here help me?

Exception has occurred: RuntimeError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
Default process group has not been initialized, please make sure to call init_process_group.
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 429, in _get_default_group
    raise RuntimeError(
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 325, in _get_group_size
    default_pg = _get_default_group()
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py", line 867, in get_world_size
    return _get_group_size(group)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/nn/modules/batchnorm.py", line 731, in forward
    world_size = torch.distributed.get_world_size(process_group)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/nn/modules/container.py", line 139, in forward
    input = module(input)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/code_lcy/FB-BEV/mmdet3d/models/fbbev/detectors/fbocc.py", line 296, in fuse_history
    feats_to_return = self.history_keyframe_time_conv(
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/mmcv/runner/fp16_utils.py", line 226, in new_func
    output = old_func(*new_args, **new_kwargs)
  File "/code_lcy/FB-BEV/mmdet3d/models/fbbev/detectors/fbocc.py", line 364, in extract_img_bev_feat
    bev_feat = self.fuse_history(bev_feat, img_metas, img[6])
  File "/code_lcy/FB-BEV/mmdet3d/models/fbbev/detectors/fbocc.py", line 386, in extract_feat
    results.update(self.extract_img_bev_feat(img, img_metas, **kwargs))
  File "/code_lcy/FB-BEV/mmdet3d/models/fbbev/detectors/fbocc.py", line 432, in forward_train
    results= self.extract_feat(
  File "/code_lcy/FB-BEV/mmdet3d/models/detectors/base.py", line 60, in forward
    return self.forward_train(**kwargs)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/mmcv/runner/fp16_utils.py", line 140, in new_func
    output = old_func(*new_args, **new_kwargs)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/mmdet/models/detectors/base.py", line 248, in train_step
    losses = self(**data)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/mmcv/parallel/data_parallel.py", line 75, in train_step
    return self.module.train_step(*inputs[0], **kwargs[0])
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/mmcv/runner/iter_based_runner.py", line 62, in train
    outputs = self.model.train_step(data_batch, self.optimizer, **kwargs)
  File "/opt/conda/envs/mmlab/lib/python3.8/site-packages/mmcv/runner/iter_based_runner.py", line 138, in run
    iter_runner(iter_loaders[i], **kwargs)
  File "/code_lcy/FB-BEV/mmdet3d/apis/train.py", line 331, in train_detector
    runner.run(data_loaders, cfg.workflow)
  File "/code_lcy/FB-BEV/mmdet3d/apis/train.py", line 357, in train_model
    train_detector(
  File "/code_lcy/FB-BEV/tools/train.py", line 360, in main
    train_model(
  File "/code_lcy/FB-BEV/tools/train.py", line 370, in <module>
    main()
  File "/opt/conda/envs/mmlab/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/conda/envs/mmlab/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/opt/conda/envs/mmlab/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/opt/conda/envs/mmlab/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/conda/envs/mmlab/lib/python3.8/runpy.py", line 194, in _run_module_as_main (Current frame)
    return _run_code(code, main_globals, None,

**Comment by 13331549708** [comment]
These should also be added to the launch.json file. Later, I asked for help and added them in and the problem was solved!
"--nnodes=1",
"--node_rank=0",
"--nproc_per_node=1",

**Comment by Orangeboy-czqzzZ** [comment]
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python Debugger: Current File",
            "type": "debugpy",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal",

            "args": ["/FB-BEV/occupancy_configs/fb_occ/fbocc-r50-cbgs_depth_16f_16x4_20e.py",
                    "--nnodes=1",
                    "--node_rank=0",
                    "--nproc_per_node=1",
            ],
            "env": {
                "PYTHONPATH": "/syncdata/onemodel/onemodel_flc_proto/1130/"
                
            }
        }
    ]
}

My  launch.json  don't work
